data_handler/handlers/loan_states/zklend/settings.py

```diff
@@ -20,70 +22,68 @@ class TokenSettings(ZkLendSpecificTokenSettings, TokenSettings):
-    "STRK": ZkLendSpecificTokenSettings(
-        collateral_factor=decimal.Decimal("0.50"),
-        debt_factor=decimal.Decimal("1"),
-        liquidation_bonus=decimal.Decimal("0.15"),
-        protocol_token_address="0x06d8fa671ef84f791b7f601fa79fea8f6ceb70b5fa84189e3159d532162efc21",
-    ),
-}
+def convert_string_to_decimal_1e27(string_big_num):
+    """
+    Method to convert big string number to decimal.
+    Example:
+    string_big_num = "800000000000000000000000000"
+    return = 0.80
+    """
+    decimal_big_num = decimal.Decimal(string_big_num)
+    scale_factor = decimal.Decimal("1e27")
+    bigResult = decimal_big_num / scale_factor
+    # Have to cast "0.800000000000000000000000000" to "800000000000000000000000000"
+    formatted_number = "{:.2f}".format(bigResult)
+    return formatted_number
 
-TOKEN_SETTINGS: dict[str, TokenSettings] = {
-    token: TokenSettings(
-        symbol=TOKEN_SETTINGS[token].symbol,
-        decimal_factor=TOKEN_SETTINGS[token].decimal_factor,
-        address=TOKEN_SETTINGS[token].address,
-        collateral_factor=ZKLEND_SPECIFIC_TOKEN_SETTINGS[token].collateral_factor,
-        debt_factor=ZKLEND_SPECIFIC_TOKEN_SETTINGS[token].debt_factor,
-        liquidation_bonus=ZKLEND_SPECIFIC_TOKEN_SETTINGS[token].liquidation_bonus,
-        protocol_token_address=ZKLEND_SPECIFIC_TOKEN_SETTINGS[
-            token
-        ].protocol_token_address,
+def get_value_by_name(data_list, name):
+    """
+    Method to return a specific value in a list based on the name.
+    """
+    return next((item['value'] for item in data_list if item['name'] == name), None)
+
+def create_and_fill_new_zklend_token_setting(reserve_data):
+    """
+    Create and fill new ZkLend token setting.
+    """
+    # Decimal Values
+    collateral_factor = convert_string_to_decimal_1e27(get_value_by_name(reserve_data, 'collateral_factor'))
+    debt_factor = decimal.Decimal("1")
+    liquidation_bonus = convert_string_to_decimal_1e27(get_value_by_name(reserve_data, 'liquidation_bonus'))
+    
+    # STR value
+    protocol_token_address = get_value_by_name(reserve_data, 'z_token_address')
+    
+    return ZkLendSpecificTokenSettings(
+        collateral_factor,
+        debt_factor,
+        liquidation_bonus,
+        protocol_token_address
+    )
+
+async def get_token_reserve_data(token_setting_address):
+    """
+    Make a call to ZKLEND_MARKET_ADDRESSES with the tokenSettingAddress (Address).
+    """
+    reserve_data = await src.blockchain_call.func_call(
+        addr=ProtocolAddresses().ZKLEND_MARKET_ADDRESSES,
+        selector="get_reserve_data",
+        calldata=[token_setting_address],
     )
-    for token in TOKEN_SETTINGS
-}
+    return reserve_data
+
+async def fetch_zklend_specific_token_settings():
+    """
+    Fetch ZkLend specific token settings.
+    """
+    # New dict to store ZkLendSpecificTokenSettings
+    NEW_ZKLEND_SPECIFIC_TOKEN_SETTINGS: dict[str, ZkLendSpecificTokenSettings] = {}
```

Comment:
This is not a constant, so use snake case.

Reply:
Fixed [0f9d33c](https://github.com/CarmineOptions/derisk-research/pull/121/commits/0f9d33cec77dcdb69c82d4f7947cde07649e2089)
pls check and let me know 😄